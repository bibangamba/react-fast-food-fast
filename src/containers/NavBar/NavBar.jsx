import React, { Component } from "react";
import {
  MDBNavbar,
  MDBNavbarBrand,
  MDBNavbarNav,
  MDBNavItem,
  MDBNavLink,
  MDBNavbarToggler,
  MDBCollapse
} from "mdbreact";

class NavigationBar extends Component {
  state = {
    collapse: false,
    isWideEnough: false
  };

  onClick = () => {
    this.setState(prevState => ({ collapse: !prevState.collapse }));
  };

  render() {
    const { collapse, isWideEnough } = this.state;
    return (
      <header>
        <MDBNavbar
          color="rgba-blue-strong"
          fixed="top"
          dark
          expand="md"
          scrolling
        >
          <MDBNavbarBrand href="/">
            <strong>FastFoodFast</strong>
          </MDBNavbarBrand>
          {!isWideEnough && <MDBNavbarToggler onClick={this.onClick} />}
          <MDBCollapse isOpen={collapse} navbar>
            <MDBNavbarNav left>
              <MDBNavItem active>
                <MDBNavLink to="/">Home</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink to="/menu">Menu</MDBNavLink>
              </MDBNavItem>
              <MDBNavItem>
                <MDBNavLink to="/orders">Orders</MDBNavLink>
              </MDBNavItem>
            </MDBNavbarNav>
          </MDBCollapse>
        </MDBNavbar>
      </header>
    );
  }
}

export default NavigationBar;
