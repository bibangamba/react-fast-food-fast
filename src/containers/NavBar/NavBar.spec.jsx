import { shallow } from "enzyme";
import React from "react";
import NavBar from "./NavBar";

const initialState = {
  collapse: false,
  isWideEnough: false
};

describe("NavBar test", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke the onClick function", () => {
    wrapper.instance().onClick();
    expect(wrapper.instance().state.collapse).toEqual(!initialState.collapse);
  });
});
