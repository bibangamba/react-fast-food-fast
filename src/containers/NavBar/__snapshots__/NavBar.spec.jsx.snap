// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar test matches snapshot 1`] = `ShallowWrapper {}`;
